### PR TITLE
fix: ASCII-fy `domainWord()`

### PR DESCRIPTION
**Pull request by schw4rzlicht**
The `en` locale includes the word "jalapeño" which is the only one with a special char. Imo this is inconsistent and it leads to other problems as well, see https://github.com/faker-js/faker/issues/1105#issuecomment-1186630351 for example.

~This PR removes the word from the locale.~

This PR uses `faker.helpers.slugify()` to remove non-ASCII characters from `domainWord()`.

### CODE REVIEW

**Comment by ST-DDT** [comment]
AFAICT: `jalapeño` is a proper English word, isn't it?
For #1105 I think we should rather ansify these characters.

**Comment by schw4rzlicht** [comment]
> AFAICT: `jalapeño` is a proper English word, isn't it?
> For #1105 I think we should rather ansify these characters.

Yes, might make more sense. I altered the Regex that filteres illegal characters in `Internet.domainWord()`, maybe that's what helps?

**Comment by matthewmayer** [comment]
if https://github.com/faker-js/faker/pull/1554 gets merged then you would be able to use the new version of `faker.helpers.slugify` to turn `jalapeño` to `jalapeno` automatically in domainWord (as domainWord seems to be a slight variant of slugify really).

**Comment by schw4rzlicht** [comment]
> if #1554 gets merged then you would be able to use the new version of `faker.helpers.slugify` to turn `jalapeño` to `jalapeno` automatically in domainWord (as domainWord seems to be a slight variant of slugify really).

From what I've seen this is right.

But that would mean:

- in order to fix that our team has to change the usage of faker for ~2000 test cases. And I don't think we're alone with this.
- we cannot call `faker.internet.domainWord()`/`faker.internet.domainName()`/`faker.internet.url()` without using `slugify()` together with it, which imo is counter intuitive, not nice to read and will definitely break in the future again, as at some point someone will forget about it
- it doesn't solve the underlying issue with `domainWord()`

As I said [here](https://github.com/faker-js/faker/issues/1105#issuecomment-1186630351):

> From what I understand, not all TLDs are even accepting internationalized domain names ([wiki](https://en.wikipedia.org/wiki/Internationalized_domain_name#Top-level_domains_accepting_IDN_registration)), so I think it is out of scope for faker to determine which are and keep track of that. Imo, domain words should just not include non-ASCII chars to keep it simple.

Additionally, internationalized domain names are converted to ASCII compatible encoding on the client when using them. They simply don't exist when building servers.

Btw, I noticed I forgot to re-add `jalapeño` after refactoring the regex of `domainWord()`. Will do that once our discussion here moves forward :)

**Comment by Shinigami92** [comment]
@schw4rzlicht I highly assume he meant using it inside faker in this PR, not in your code to wrap faker calls.

**Comment by matthewmayer** [comment]
right could you change domainWord to just be something like

```
domainWord(): string {
    return this.faker.helpers
      .slugify(`${this.faker.word.adjective()}-${this.faker.word.noun()}`)
      .toLowerCase();
  }
```

**Comment by schw4rzlicht** [comment]
> @schw4rzlicht I highly assume he meant using it inside faker in this PR, not in your code to wrap faker calls.

Ah alright, misunderstood that, sorry 🤦

Thanks @matthewmayer, I will update the PR ~today or tomorrow~ :)

EDIT: Wasn't able to do it yet, too much work on other projects. Will do asap though.

File: src/modules/internet/index.ts [hunk 1]
```diff
@@ -358,8 +358,8 @@ export class InternetModule {
    * @since 2.0.1
    */
   domainWord(): string {
-    return `${this.faker.word.adjective()}-${this.faker.word.noun()}`
-      .replace(/([\\~#&*{}/:<>?|\"'])/gi, '')
+    return this.faker.helpers
+      .slugify(`${this.faker.word.adjective()}-${this.faker.word.noun()}`)
       .replace(/\s/g, '-')
       .replace(/-{2,}/g, '-')
```
Comment (matthewmayer) [inline src/modules/internet/index.ts]: I don't think you need these two lines any more as slugify already converts spaces to - for you

**Review by matthewmayer — CHANGES_REQUESTED** [review]

**Review by ST-DDT — APPROVED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Review by ejcheng — APPROVED** [review]

**Pull request merged**